### PR TITLE
Bring compilation down to ~4s

### PR DESCRIPTION
**Pull request by benesch**
See individual commits for details!

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/1477)

### CODE REVIEW

**Comment by jseldess** [comment]
Don't see why this would have anything to do with your changes, but the hubspot email sign-up at the top of each release notes page is missing:

screen shot 2017-05-31 at 12 43 46 pm

Compare that to the live page:

screen shot 2017-05-31 at 12 44 00 pm

**Comment by jseldess** [comment]
A few minor issues:

- On [Install Client Drivers](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4f772439b1bef61bec7514063e99f1548e91592/install-client-drivers.html), the html comment is now visible for some reason.
- On [Constraints](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4f772439b1bef61bec7514063e99f1548e91592/constraints.html), some of the code blocks aren't getting indented properly.
- On [Data Types](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4f772439b1bef61bec7514063e99f1548e91592/data-types.html), the main table is broken, and the green tip isn't indented properly.
- On [`SERIAL`](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4f772439b1bef61bec7514063e99f1548e91592/serial.html#auto-incrementing-is-not-always-sequential), the number and indentation are off here.
- More code block and callout indentation issues on [`COLLATE`](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4f772439b1bef61bec7514063e99f1548e91592/collate.html).
- On [Recommended Production Settings](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4f772439b1bef61bec7514063e99f1548e91592/recommended-production-settings.html), some anchor links, steps, and code blocks aren't rendering.

**Comment by sploiselle** [comment]
Railroad diagrams are broken ([example](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4f772439b1bef61bec7514063e99f1548e91592/begin-transaction.html))

File: _plugins/template_cache.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module TemplateWithCaching
```
Comment (benesch) [inline _plugins/template_cache.rb]: I ended up just nuking that module; causing more problems than it solved.

**Comment by benesch** [comment]
Ok, I think I've fixed all the bugs you guys found! PTAL.

**Comment by benesch** [comment]
This is finally green! 🎉

**Comment by benesch** [comment]
Eek, yep, my fault. Let me see if I can appease TeamCity so it works with and without this PR.

**Comment by benesch** [comment]
Ok! Ready when you are, @sploiselle.

**Comment by sploiselle** [comment]
Redcarpet's breaking on nested code blocks, e.g., [Cluster Setup Troubleshooting](http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/8b4fe1c7b4f92851a9d0922d43038d92d07300fe/cluster-setup-troubleshooting.html#1-start-a-single-node-cluster).

Can we resolve this with changes to the markdown, or is it a Redcarpet issue?

**Comment by sploiselle** [comment]
Didn't realize my last comment was already tracked in #1492; do you want to do a fast-follow PR to fix these issues @jseldess? Otherwise, I'm glad to take it

**Comment by jseldess** [comment]
@benesch, Sean found another page where the indentation needs to be fixed up. Would you handle that real quick before merging? Otherwise, yeah, would be great if you'd follow up with a fix after merge, @sploiselle.

**Comment by benesch** [comment]
Oh, good catch! Should be fixed now.

**Comment by sploiselle** [comment]
You are such a champ, @benesch

**Comment by benesch** [comment]
🍻

**Pull request merged**